inspirehep/utils/tickets.py

```diff
@@ -24,26 +24,234 @@
+
+    :param queue: where the ticket will be created
+    :type queue: string
+
+    :param requestors: username to set to requestors field of the ticket
+    :type requestors: string
+
+    :param template_path: path to the template for the ticket body
+
+    :param template_context: context object to be used to render template
+
+    :param recid: record id to be set custom RecordID field
+    :type recid: integer
+
+    :param kwargs: Other arguments possible to set:
+
+                Cc, AdminCc, Owner, Status,Priority, InitialPriority,
+                FinalPriority, TimeEstimated, Starts, Due,
+                ... (according to RT fields)
+
+                Custom fields CF.{<CustomFieldName>} could be set
+                with keywords CF_CustomFieldName.
+    :returns: ID of the new ticket or ``-1``, if it fails
+    :rtype: integer
+    """
+    body = render_template(template_path, **template_context).strip()
+    subject = template_context.get("subject")
+    return create_ticket(queue, requestors, body, subject, recid, **kwargs)
+
+
+def resolve_ticket(ticket_id):
+    """Resolves the given ticket
+
+    :type ticket_id: integer
+    """
+    try:
+        rt_instance.edit_ticket(
+            ticket_id=ticket_id,
+            Status="resolved",
+        )
+    except IndexError:
+        # Raise exception only ticket isn't already resolved
+        ticket = rt_instance.get_ticket(ticket_id)
+        if ticket["Status"] != "resolved":
+            raise EditTicketException()
+
+
+def get_queues():
+    """Returns list of all queues as {id, name} dict
+
+    :rtype: dict - with ``name (string)``, ``id (integer)`` properties
+    """
+    return _get_all_of("queue")
+
+
+def get_users():
+    """Returns list of all users as {id, name} dict
+
+    :rtype: dict - with ``name (string)``, ``id (integer)`` properties
+    """
+    return _get_all_of("user")
+
+
+def _get_all_of(query_type):
+    """Utility function to share the code for performing custom get all requests
+     and parsing the result
+
+    :param query_type: the type of quer, either ``'queue'`` or ``'user'``
+
+    :rtype: dict - with ``name (string)``, ``id (integer)`` properties
+    """
+    query_suffix = query_type + "?query="
+    url = join(rt_instance.url, "search", query_suffix)
+    response = rt_instance.session.get(url)
+    raw_result = response.content.decode(response.encoding.lower())
+    # parse raw result
+    lines = raw_result.split("\n")
+    # remove status and empty lines
+    del lines[:2]
+    del lines[-3:]
+    # create dict for each result item
+    return map(_query_result_item_id_name_mapper, lines)
+
+
+def _query_result_item_id_name_mapper(raw_item):
+    """Mapper function
+
+    Takes a string like ``'17: CoolUser'`` and
+    returns ``{'id': '17', 'name': 'CoolUser'}``
+    """
+    parts = raw_item.split(": ")
+    return dict(id=parts[0], name=parts[1])
+
+
+def _strip_lines(multiline_string):
+    return "\n ".join(
```

Comment:
Actually that's was workaround before, I am not sure what the put in the doc string that someone can't understand from the code.